modules/google-analytics/wp-google-analytics.php

```diff
@@ -163,6 +169,136 @@ private function _get_tracking_code() {
 
 		return '';
 	}
+
+	/**
+	 * Used to filter in the anonymize IP snippet to the custom vars array for classic analytics
+	 * Ref https://developers.google.com/analytics/devguides/collection/gajs/methods/gaJSApi_gat#_gat._anonymizelp
+	 * @param array custom vars to be filtered
+	 * @return array possibly updated custom vars
+	 */
+	public function jetpack_wga_classic_anonymize_ip( $custom_vars ) {
+		$o = get_option( 'jetpack_wga' );
+		$anonymize_ip = isset( $o[ 'anonymize_ip' ] ) ? $o[ 'anonymize_ip' ] : false;
+		if ( $anonymize_ip ) {
+			array_push( $custom_vars, "_gaq.push(['_gat._anonymizeIp']);" );
+		}
+
+		return $custom_vars;
+	}
+
+	/**
+	 * Used to filter in the order details to the custom vars array for classic analytics
+	 * @param array custom vars to be filtered
+	 * @return array possibly updated custom vars
+	 */
+	public function jetpack_wga_classic_track_purchases( $custom_vars ) {
+		global $wp;
+
+		// Ref: https://developers.google.com/analytics/devguides/collection/gajs/gaTrackingEcommerce#Example
+		$o = get_option( 'jetpack_wga' );
+		$ec_track_purchases = isset( $o[ 'ec_track_purchases' ] ) ? $o[ 'ec_track_purchases' ] : false;
+		$minimum_woocommerce_active = class_exists( 'WooCommerce' ) && version_compare( WC_VERSION, '3.0', '>=' );
+		if ( $ec_track_purchases && $minimum_woocommerce_active && is_order_received_page() ) {
+			$order_id = isset( $wp->query_vars['order-received'] ) ? $wp->query_vars['order-received'] : 0;
+			if ( 0 < $order_id && 1 != get_post_meta( $order_id, '_ga_tracked', true ) ) {
+				$order = new WC_Order( $order_id );
+
+				// [ '_add_Trans', '123', 'Site Title', '21.00', '1.00', '5.00', 'Snohomish', 'WA', 'USA' ]
+				$code = "_gaq.push(['_addTrans',
```

Comment:
Instead of concatenating and relying on `esc_js` here, let's instead use `json_encode` here?

```
array_push(
    $custom_vars,
    sprintf(
        '_gaq.push( %s )', 
        json_encode(
            array( 'one', 'two', 'three' )
        )
    )
);
```
This will make sure nothing will accidentally or intentionally break us out of context.

Reply:
Fixed in https://github.com/Automattic/jetpack/pull/7817/commits/c531b0096047134db8f7aa71a8d924e8a53a23d1